builtin/builtin.go

```diff
@@ -826,6 +826,50 @@ object.
 The globals and locals are dictionaries, defaulting to the current
 globals and locals.  If only globals is given, locals defaults to it.`
 
+const isinstance_doc = `isinstance(obj, class_or_tuple) -> bool
+
+Return whether an object is an instance of a class or of a subclass thereof.
+
+A tuple, as in isinstance(x, (A, B, ...)), may be given as the target to
+check against. This is equivalent to isinstance(x, A) or isinstance(x, B)
+or ... etc.
+`
+
+func isinstance(args py.Tuple) (py.Bool, error) {
+	switch args[1].(type) {
+	case py.Tuple:
+		{
+			var res py.Bool
+			var class = args[1].(py.Tuple)
+			for idx := range class {
+				var new_args []py.Object
+				new_args = append(new_args, args[0], class[idx])
+				temp, err := isinstance(new_args)
+				if err != nil {
+					return false, err
+				}
+				res = res || temp
```

Comment:
This should say

```go
if res {
    return res
}
```

to short circuit the rest of the evaluations shouldn't it?

If you do that you can move the definition of res to here.